client/src/components/Ranks/TopRanks.js

```diff
@@ -1,14 +1,111 @@
 import React from 'react'
-
+import { useState , useEffect} from 'react';
+import AnimeCard from './cards/AnimeCard';
```

Comment:
I appreciate you split this out this way, with one main container card delegating to card components of different types. This is way clearer and I understand what you were doing immediately. You did a good job with writing this part. Making it so other people can understand what you're doing is a big part of the battle!